src/Microsoft.DotNet.Cli.Utils/Command.cs

```diff
@@ -100,7 +102,11 @@ private Command(CommandSpec commandSpec)
                 outputPath: outputPath,
                 applicationName: applicationName);
 
-            if (commandSpec == null)
+            if (_knownCommandsAvailableAsDotNetTool.Contains(commandName, StringComparer.InvariantCultureIgnoreCase) && commandSpec == null)
```

Comment:
Terms of `&&` should be re-ordered since there's no need to do N string comparison if command was already found.

Also consider hoisting the commandSpec == null to a single check.

Reply:
hoisted. And added extra nesting. Still looks clean